dpctl/tensor/libtensor/include/kernels/sorting/radix_sort.hpp

```diff
@@ -0,0 +1,1921 @@
+        sort_ev = exec_q.submit([=](sycl::handler &cgh) {
+            cgh.depends_on(sort_ev);
+
+            const sycl::context &ctx = exec_q.get_context();
+
+            using dpctl::tensor::alloc_utils::sycl_free_noexcept;
+            cgh.host_task([ctx, count_ptr, tmp_arr]() {
+                sycl_free_noexcept(tmp_arr, ctx);
+                sycl_free_noexcept(count_ptr, ctx);
+            });
+        });
+    }
+
+    return sort_ev;
+}
+
+struct IdentityProj
+{
+    constexpr IdentityProj() {}
+
+    template <typename T> constexpr T operator()(T val) const { return val; }
+};
+
+template <typename ValueT, typename IndexT> struct ValueProj
+{
+    constexpr ValueProj() {}
+
+    constexpr ValueT operator()(const std::pair<ValueT, IndexT> &pair) const
+    {
+        return pair.first;
+    }
+};
+
+template <typename IndexT, typename ValueT, typename ProjT> struct IndexedProj
+{
+    IndexedProj(const ValueT *arg_ptr, const ProjT &proj_op)
+        : ptr(arg_ptr), value_projector(proj_op)
+    {
+    }
+
+    auto operator()(IndexT i) const { return value_projector(ptr[i]); }
+
+private:
+    const ValueT *ptr;
+    ProjT value_projector;
+};
+
+} // end of namespace radix_sort_details
+
+template <typename argTy>
+sycl::event
+radix_sort_axis1_contig_impl(sycl::queue &exec_q,
+                             const bool sort_ascending,
+                             // number of sub-arrays to sort (num. of rows in a
+                             // matrix when sorting over rows)
+                             size_t iter_nelems,
+                             // size of each array to sort  (length of rows,
+                             // i.e. number of columns)
+                             size_t sort_nelems,
+                             const char *arg_cp,
+                             char *res_cp,
+                             ssize_t iter_arg_offset,
+                             ssize_t iter_res_offset,
+                             ssize_t sort_arg_offset,
+                             ssize_t sort_res_offset,
+                             const std::vector<sycl::event> &depends)
+{
+    const argTy *arg_tp = reinterpret_cast<const argTy *>(arg_cp) +
+                          iter_arg_offset + sort_arg_offset;
+    argTy *res_tp =
+        reinterpret_cast<argTy *>(res_cp) + iter_res_offset + sort_res_offset;
+
+    using Proj = radix_sort_details::IdentityProj;
+    constexpr Proj proj_op{};
+
+    sycl::event radix_sort_ev =
+        radix_sort_details::parallel_radix_sort_impl<argTy, Proj>(
+            exec_q, iter_nelems, sort_nelems, arg_tp, res_tp, proj_op,
+            sort_ascending, depends);
+
+    return radix_sort_ev;
+}
+
+template <typename ValueT, typename IndexT>
+class populate_indexed_data_for_radix_sort_krn;
+
+template <typename ValueT, typename IndexT>
+class index_write_out_for_radix_sort_krn;
+
+template <typename argTy, typename IndexTy>
+sycl::event
+radix_argsort_axis1_contig_impl(sycl::queue &exec_q,
```

Comment:
This impl function seems to be unused anywhere. Should it be kept around or removed as dead code?

Reply:
Good catch, @ndgrigorian ! I am going to remove it